### DISCUSSION TITLE
BleskOS64

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#26] [Ideas]
Hello Vendelín,

I was unable to boot BleskOS from a wize thin client (https://www.parkytowers.me.uk/thin/wyse/z/z90d7/).
It is complaining that it can find anything to boot from UEFI64.

Do you plan to add EFI and 64bit capabilities?

More and more +10 years old 64bit cpu are now not enough powerfull for even Linux :-(.

**Comment by VendelinSlezak** [comment]
BleskOS is 32 bit system, so it will work on any 64 bit computer. Only problem is that BleskOS now support only Legacy boot and most of 64 bit computers have UEFI as default. Good news is that most of computers with UEFI(especially older ones) support also Legacy boot. So I would recomment you to check BIOS settings if there is some option to change boot from EFI to Legacy and if yes, change it to Legacy and try to run BleskOS again.

For now I am not planning to add EFI capability, because really almost all computers support Legacy boot. Maybe some day in future.

And I also now do not plan to add 64 bit capability, because BleskOS as 32 bit system will run also on all 64 bit computers. So only real improvement would be that 64 bit allow you to access more memory than 4 GB. However, because BleskOS is very small system, it need only around 16 - 32 MB for itself and rest of memory is for user files. So unless you are working with really, really big files, even 512 MB memory will be comfortably enough to use BleskOS.

**Comment by defdefred** [comment]
Make sens. I will have to find a way to break the password locking the thin client BIOS :-)

64 bit asm could be useful to make efficient library or driver, like video decoding without using the GPU...

I use Ventoy to boot the BleskOS iso and it seems that the tools is taking care of some EFI needs already.

**Comment by VendelinSlezak** [comment]
"I will have to find a way to break the password locking the thin client BIOS"
I think safest way is to try to remove CMOS baterry for at least 30 seconds to reset all BIOS settings. But of course it depends if you can use this computer for experimenting.

"64 bit asm could be useful to make efficient library or driver, like video decoding without using the GPU..."
Yes, 64 bits could be used for faster operations, but now I am grateful that even 32 bit code work :)

**Comment by defdefred** [comment]
Did you know that 32bit x86 are still produced with 1Ghz speed?
https://www.compactpc.com.tw/products/lists/2

I wonder if BleskOS is already doing multitasking and multiprocessor?

**Comment by VendelinSlezak** [comment]
> I wonder if BleskOS is already doing multitasking and multiprocessor?

No, BleskOS now is one big single task which have all "programs" inside. So it is some "pseudomultitasking" because from user perspective it looks like there are multiple programs running, but in reality there is only one "program" that makes everything.

**Comment by defdefred** [comment]
So it is an omnipotent application running baremetal :-).

Could we have several part of the application running in parallel?
Let's say some music playing with typing text?

**Comment by VendelinSlezak** [comment]
Yes, of course, it only depends how it is programmed :) Code for playing music on background is already in BleskOS. I only need to add code to use it to play music in background in other “applications”
